### DISCUSSION TITLE
hitsPerPage to 0 when search box is empty

### DISCUSSION THREAD

**Discussion by anjanesh** [algolia/instantsearch#5977] [Q&A]
I have :
```
instantsearch.widgets.configure({ hitsPerPage: 8 }),
```
But when the search page is loaded it shows a default list of 8 results for an empty search.

I'm using Vanilla JavaScript. How do I show 8 results when only searched for and nothing when search box is empty ?

**Comment by Haroenv** [comment]
Do you want to create an autocomplete instead of an InstantSearch experience? I recommend you have a look at the autocomplete library: https://github.com/algolia/autocomplete

**Reply by anjanesh** [reply]
I found what I was looking for - https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-display/js/#handling-empty-queries
